docs/site/todo-list-tutorial-sqldb.md

```diff
@@ -0,0 +1,133 @@
+---
+lang: en
+title: 'Running on relational database'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/todo-list-tutorial-sqldb.html
+summary:
+  LoopBack 4 TodoList Application Tutorial - Running on Relational Database
+---
+
+If you are running this example using a relational database, here are the extra
+steps. PostgreSQL is used for illustration, it works in a similar way in other
+relational databases.
+
+### Create a new datasource
+
+Run `lb4 datasource`. Specify the datasource name as `psqlds` and select
+`PostgreSQL` as the connector.
+
+```sh
+$ lb4 datasource
+? Datasource name: psqlds
+? Select the connector for psqlds:
+  Redis key-value connector (supported by StrongLoop)
+  MongoDB (supported by StrongLoop)
+  MySQL (supported by StrongLoop)
+❯ PostgreSQL (supported by StrongLoop)
+  Oracle (supported by StrongLoop)
+  Microsoft SQL (supported by StrongLoop)
+  REST services (supported by StrongLoop)
+  ...
+```
+
+### Update the Repository classes using the new datasource
+
+The Repository classes are used to bind the datasource and the models. Since
+we've changed to another datasource, we need to update the repositories which
+are located in the `src\repositories` folder.
+
+In `todo.repository.ts`, inject the new datasource in the constructor as below:
+
+```ts
+constructor(
+    @inject('datasources.psqlds') dataSource: juggler.DataSource,
```

Comment:
IIUC, as a result of this tutorial step, we end up with two datasources in our project:
 - `db` datasource backed by `memory` connector - NOT USED BY ANY REPO
 - `psqlds` datasource backed by `postgresql` connector - USED EVERYWHERE

I don't like the idea of having a datasource that's not used by any repository.

Also I consider it as a bad practice to name artifacts after implementation details. It's better to name them after our intention & their function. The datasource called `db` is meant to be the primary data storage used by the application.

Can we find a way how to preserve `db` datasource name please?

For example, we can tell users to simply change the configuration of the existing `db` datasource to use `postgresql` instead of `memory` connector.

If we want a more interactive approach that will prompt them for PostgreSQL config, then perhaps we can tell them to `rm src/datasources/db.datasource.*` and run `lb4 datasource db` to re-create it with different config.